src/java/org/apache/cassandra/repair/autorepair/AutoRepairUtils.java

```diff
@@ -447,6 +465,270 @@ public static AutoRepairHistory getHostWithLongestUnrepairTime(RepairType repair
+                        }
+                    }
+                }
+            }
+        }
+        return hostsBeingRepaired;
+    }
+
+    /**
+     * Identifies the most eligible host to repair for nodes preceding or equal to this nodes' lastRepairFinishTime.
+     * The criteria for this is to find the node with the oldest last repair finish time of which none of its replicas
+     * are currently under repair.
+     * @return The most eligible host to repair or null if no candidates before and including this nodes' current repair status.
+     */
+    @VisibleForTesting
+    public static AutoRepairHistory getMostEligibleHostToRepair(RepairType repairType, CurrentRepairStatus currentRepairStatus, UUID myId)
+    {
+        // 0. If this repairType allows parallel replica repair, short circuit and return the host with the longest unrepair time
+        AutoRepairConfig config = AutoRepairService.instance.getAutoRepairConfig();
+        if (config.getAllowParallelReplicaRepair(repairType))
+        {
+            return getHostWithLongestUnrepairTime(currentRepairStatus.historiesWithoutOnGoingRepair);
+        }
+
+        // 1. Sort repair histories from oldest completed to newest
+        Stream<AutoRepairHistory> finishedRepairHistories = currentRepairStatus.historiesWithoutOnGoingRepair
+                                                            .stream()
+                                                            .sorted((history0, history1) -> {
+                                                                if (history0.lastRepairFinishTime == history1.lastRepairFinishTime)
+                                                                    return 0;
+                                                                else
+                                                                    return history0.lastRepairFinishTime < history1.lastRepairFinishTime ? -1 : 1;
+                                                            });
+
+        // 2. Optimization: Truncate repair histories after myId so we don't evaluate anything more recent as if we
+        // aren't interested in anything that isn't this node.
+        final AtomicBoolean myHistoryFound = new AtomicBoolean(false);
+        finishedRepairHistories = finishedRepairHistories.takeWhile((history) -> {
+            if (!myHistoryFound.get())
+            {
+                if (history.hostId.equals(myId))
+                {
+                    myHistoryFound.set(true);
+                }
+                return true;
+            }
+            return false;
+        });
+
+        // If there are any hosts with ongoing repair, filter the repair histories to not include nodes whose replicas
+        // are ongoing repair.
+        Map<UUID, RepairType> hostsBeingRepairedToRepairType = getHostsBeingRepaired(repairType, currentRepairStatus);
+
+        // 3. If I am already actively being repaired in another schedule, defer submitting repears;  If the repair type
```

Comment:
did you mean repairs maybe?
```suggestion
        // 3. If I am already actively being repaired in another schedule, defer submitting repairs;  If the repair type
```

Reply:
re🍐 's are delicious!  Yeah spelled this wrong, will fix, thank you!